src/server/scripts/Draenor/zone_draenor_shadowmoon_valley.cpp

```diff
@@ -110,7 +108,25 @@ struct npc_baros_alexston : public ScriptedAI
     }
 };
 
+// 160938 - Despawn All Summons (Garrison Intro Only)
+class spell_despawn_all_summons_garrison_intro_only : public SpellScript
+{
+    PrepareSpellScript(spell_despawn_all_summons_garrison_intro_only);
+
+    void HandleScript(SpellEffIndex /*effIndex*/)
+    {
+        if (GetHitCreature())
```

Comment:
You need to check if the creatures being used are the player's summons, other players may not see other players' personal summons, but the core can and it might wrongly delete other players' summons if the entries match:

if (GetHitCreature() && GetHitCreature()->GetSummoner() == GetCaster())
GetHitCreature()->DespawnOrUnsummon();

Reply:
youre right ty